src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -137,6 +122,51 @@ private void raiseSQLExceptionIfConnectionIsClosed() throws SQLException {
     }
   }
 
+  /**
+   * Function to send in-band telemetry data about DatabaseMetadata get API calls and their
+   * associated SHOW commands
+   *
+   * @param resultSet The ResultSet generated from the SHOW command in the function call. Can be of
+   *     type SnowflakeResultSet or SnowflakeDatabaseMetaDataResultSet
+   * @param functionName name of DatabaseMetadata API function call
+   * @param catalog database
+   * @param schema schema
+   * @param generalNamePattern name of table, function, etc
+   * @param specificNamePattern name of table column, function parameter name, etc
+   */
+  private void sendInBandTelemetryMetadataMetrics(
+      ResultSet resultSet,
+      String functionName,
+      String catalog,
+      String schema,
+      String generalNamePattern,
+      String specificNamePattern) {
+    String queryId = "";
+    if (resultSet instanceof SnowflakeResultSet) {
+      try {
+        queryId = resultSet.unwrap(SnowflakeResultSet.class).getQueryID();
+      } catch (SQLException e) {
+        // do nothing here; just don't include the query ID
+      }
+    } else if (resultSet instanceof SnowflakeDatabaseMetaDataResultSet) {
+      try {
+        queryId = resultSet.unwrap(SnowflakeDatabaseMetaDataResultSet.class).getQueryID();
+      } catch (SQLException e) {
+        // do nothing here; just don't include the query ID
+      }
+    }
+    ObjectNode ibValue = mapper.createObjectNode();
+    ibValue.put("type", TelemetryField.METADATA_METRICS.toString());
```

Comment:
@sfc-gh-stakeda the SHOW command has been taken out of the message. Now it is query ID, function name, and function parameters.